src/libraries/Common/src/System/Net/Security/SslClientAuthenticationOptionsExtensions.cs

```diff
@@ -30,7 +30,7 @@ public static SslClientAuthenticationOptions ShallowClone(this SslClientAuthenti
 
 #if DEBUG
             // Try to detect if a property gets added that we're not copying correctly.
-            foreach (PropertyInfo pi in options.GetType().GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly))
+            foreach (PropertyInfo pi in typeof(SslClientAuthenticationOptions).GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly))
```

Comment:
Maybe also:
```C#
Debug.Assert(typeof(SslClientAuthenticationOptions) == options.GetType());
```
(assuming that itself doesn't trigger a linker warning).

Reply:
> assuming that itself doesn't trigger a linker warning

Just calling `GetType()` won't trigger a warning. It's only when you start grabbing things off the Type and don't ensure they aren't trimmed that will cause warnings.

Reply:
Right, that's why I'm suggesting asserting the type, because otherwise it's broken.

Reply:
Oh, you're saying a publicly provided instance can currently find its way in here?  If so, then yeah, we need to do something additional as well, e.g. remove DeclaredOnly.

Reply:
That said, this is all debug-only code, so it impacts no one other than someone developing System.Net.Http itself.

Reply:
So then I guess the way to do this would be to keep the typeof (as opposed to the GetType() call) and only remove the DeclaredOnly flags?

Reply:
That's fine for now.  Unrelated to this PR, we should probably add a test to make sure that things don't blow up if someone passes in a derived type.